contracts/interchain-token-service/src/state.rs

```diff
@@ -60,24 +79,25 @@ pub enum MessageDirection {
 
 /// The deployment type of the token.
 pub enum TokenDeploymentType {
-    /// The token is trustless, i.e only owned by ITS.
+    /// The token is trustless, i.e. only owned by ITS.
     Trustless,
     /// The token has a custom minter.
     CustomMinter,
 }
 
 /// Global config for a token.
 #[cw_serde]
-pub struct TokenConfig {
+pub struct GlobalTokenConfig {
+    pub token_id: TokenId,
     pub origin_chain: ChainNameRaw,
 }
 
 const CONFIG: Item<Config> = Item::new("config");
 const ITS_CONTRACTS: Map<&ChainNameRaw, Address> = Map::new("its_contracts");
 const CHAIN_CONFIGS: Map<&ChainNameRaw, ChainConfig> = Map::new("chain_configs");
-const TOKEN_CHAIN_INFO: Map<&(ChainNameRaw, TokenId), TokenChainInfo> =
-    Map::new("token_chain_info");
-const TOKEN_CONFIGS: Map<&TokenId, TokenConfig> = Map::new("token_configs");
+const TOKEN_INSTANTIATIONS: Map<&(ChainNameRaw, TokenId), TokenInstance> =
```

Comment:
```suggestion
const TOKEN_INSTANTIATIONS: Map<(&ChainNameRaw, &TokenId), TokenInstance> =
```
Can we switch to this to avoid some cloning?

Reply:
I'll have to check if the key needs to get deserialized when you read a value. If so your suggestion wouldn't work because of lifetimes. Otherwise I'll change it

Reply:
doesn't work